api/server/router/container/container_routes.go

```diff
@@ -619,6 +628,14 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 		}
 	}
 
+	if hostConfig.PidsLimit != nil && *hostConfig.PidsLimit <= 0 {
+		// Don't set a limit if either no limit was specified, or "unlimited" was
+		// explicitly set.
+		// Both `0` and `-1` are accepted as "unlimited", and historically any
+		// negative value was accepted, so treat those as "unlimited" as well.
+		hostConfig.PidsLimit = nil
+	}
+
```

Comment:
Funny how this is different in `postContainerUpdate`

https://github.com/moby/moby/blob/e36260f39abd1def26456ac43c6a8f77336523b2/api/server/router/container/container_routes.go#L456-L462